fastestimator/schedule/schedule.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+
+class Scheduler:
+    def get_current_value(self, epoch: int):
+        return None
+
+    def get_items(self):
+        raise NotImplementedError
+
+
+class RepeatScheduler(Scheduler):
+    def __init__(self, repeat_list):
+        assert isinstance(repeat_list, list), "must provide a list as input of RepeatSchedule"
+        self.repeat_list = repeat_list
+        self.cycle_length = len(repeat_list)
+        assert self.cycle_length > 1, "list length must be greater than 1"
+
+    def get_current_value(self, epoch: int):
+        return self.repeat_list[epoch % self.cycle_length]
+
+    def get_items(self):
+        return self.repeat_list
+
+
+class FunctionScheduler(Scheduler):
+    def __init__(self, schedule_fun):
+        assert hasattr(schedule_fun, '__call__'), "must provide a lambda function with epoch as input"
+        self.schedule_fun = schedule_fun
+
+    def get_current_value(self, epoch: int):
+        return self.schedule_fun(epoch)
+
+
+class EpochScheduler(Scheduler):
+    def __init__(self, epoch_dict):
+        assert isinstance(epoch_dict, dict), "must provide dictionary as epoch_dict"
+        self.epoch_dict = epoch_dict
+        self.keys = sorted(self.epoch_dict)
+        assert 0 in self.epoch_dict, "epoch 0 is missing in dictionary, use None if no op is needed"
```

Comment:
why are we enforcing this rather than just adding it implicitly ourselves like before?

Reply:
to make life easier and more transparent.   it will also avoid making mistakes